lib/puppet/defaults.rb

```diff
@@ -1395,7 +1395,24 @@ def self.default_diffargs
     },
     :server => {
       :default => "puppet",
-      :desc => "The puppet master server to which the puppet agent should connect."
+      :desc => "The puppet master server to which the puppet agent should connect.",
+      :hook => proc { |value|
+          if Puppet.settings.set_by_config?(:server_list)
```

Comment:
`--server` is used in many places outside of the puppet repo. There are a bunch of github results:

[bash](https://github.com/search?l=bash&q=%22puppet+agent+--server%22&type=Code&utf8=%E2%9C%93)
[ruby](https://github.com/search?l=ruby&q=%22puppet+agent+--server%22&type=Code&utf8=%E2%9C%93)

One one hand, some of those results are forks of puppet. But on the other hand, it's only looking for cases where --server follows agent, so not `puppet agent -t --server`.

Also it doesn't include all of the puppet, PE and modules acceptance tests that would need to be updated to remove the deprecation. There's also MCO, though we just removed the --server option by default.

There's also customer integration/workflows to consider, e.g. running puppet from cron, etc and documentation:

http://shapeshed.com/connecting-clients-to-a-puppet-master/
https://www.unixmen.com/puppet-agent-install-and-configure-to-all-linuxunix/
http://www.juniper.net/techpubs/en_US/junos-puppet2.0/topics/task/installation/automation-junos-puppet-agent-installing.html